src/Features/Core/Portable/DesignerAttribute/DesignerAttributeDiscoveryService.cs

```diff
@@ -305,6 +324,33 @@ private static async ValueTask<bool> HasDesignerCategoryTypeAsync(Project projec
 
         return null;
 
+        static IEnumerable<ITypeSymbol> GetBaseTypesAndThis(Compilation compilation, INamedTypeSymbol firstType)
+        {
+            var current = firstType;
+            while (current != null)
+            {
+                yield return current;
+                current = current.BaseType;
+
+                if (current is IErrorTypeSymbol errorType)
+                    current = TryMapToNonErrorType(compilation, errorType);
+            }
+        }
+
+        static INamedTypeSymbol? TryMapToNonErrorType(Compilation compilation, IErrorTypeSymbol errorType)
+        {
+            foreach (var wellKnownNamespace in s_wellKnownDesignerNamespaces)
+            {
+                var wellKnownType = compilation.GetTypeByMetadataName($"{wellKnownNamespace}.{errorType.Name}");
```

Comment:
🤔When will this happen to user? Namespace misspelling, why the namespace in SemanticModel is wrong?

Reply:
Basically, as a restore happens, we can lose and then get references again.   Also, generators may not run again until we know about them again, so we lost global usings. 

This keeps everything still looking good (and usable for the designer side) while we get back into a good state again.